### DISCUSSION TITLE
Serverless function error

### DISCUSSION THREAD

**Discussion by FlyingTurkman** [vercel/community#3133] [Help]
### Summary

I have been deploy an app and its downing sometimes with serverless function error.
Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"TypeError: fetch failed","reason":{"errorType":"TypeError","errorMessage":"fetch failed","cause":{"errorType":"HeadersTimeoutError","errorMessage":"Headers Timeout Error","code":"UND_ERR_HEADERS_TIMEOUT","name":"HeadersTimeoutError","message":"Headers Timeout Error","stack":["HeadersTimeoutError: Headers Timeout Error","    at Timeout.onParserTimeout [as _onTimeout] (/var/task/node_modules/next/dist/compiled/undici/index.js:1:60643)","    at listOnTimeout (node:internal/timers:571:11)","    at process.processTimers (node:internal/timers:512:7)"]},"stack":["TypeError: fetch failed","    at Object.fetch (node:internal/deps/undici/undici:11457:11)","    at async /var/task/.next/server/pages/_app.js:1916:24"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: TypeError: fetch failed","    at process.<anonymous> (file:///var/runtime/index.mjs:1189:17)","    at process.emit (node:events:525:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
Unknown application error occurred
Runtime.Unknown
Its error log
![image](https://github.com/vercel/community/assets/99084104/a0566b5a-6573-4e8b-ba80-59e317478fcb)

### Steps to Reproduce

![image](https://github.com/vercel/community/assets/99084104/44ad3181-81d8-4b04-a4fe-ffa9ed0c97e3)

**Comment by ali-habibzadeh** [comment]
same

**Comment by FlyingTurkman** [comment]
@ali-habibzadeh I figured it out. I am spliting server and client components. For example


server component
```tsx
import PageView from './PageView'
export default async function Page() {
const data = await fetch('...')

  return(
    <main>
      <PageView data={data}/>
    </main>
  )
}
```
client component
```tsx
'use client'
export default function PageView({ data }: { data: dataType ) {
  return(
    <div>
      data.map((d) => {
        return(
        <div>{d}</div>
        )
      })
    </div>
  )
}
```

like that
